Inc/HALALMock/Services/ADC/ADC.hpp

```diff
@@ -22,24 +23,31 @@ using std::string;
 #define LPTIM3_PERIOD 6875
 
 #define ADC_MAX_VOLTAGE 3.3
-#define MAX_12BIT 4095.0
-#define MAX_16BIT 65535.0
+#define MAX_16BIT ((1<<16) - 1)
+#define MAX_14BIT ((1<<14) - 1)
+#define MAX_12BIT ((1<<12) - 1)
+#define MAX_10BIT ((1<<10) - 1)
```

Comment:
careful, this will probably be considered an int by the compiler and it might give you some nasty unexpected results